src/webots/nodes/WbTransform.cpp

```diff
@@ -52,6 +60,24 @@ WbTransform::~WbTransform() {
     wr_node_delete(WR_NODE(wrenNode()));
 }
 
+void WbTransform::reset(const QString &id) {
+  WbGroup::reset(id);
+  // note: for solids, the setting of these parameters has to occur only if mJointParents.size() == 0 and is taken care of in
+  // WbSolid::reset, otherwise it breaks the resetting of hinge based joints
+  if (nodeModelName() != "TrackWheel" && nodeModelName() != "Solid") {
+    setTranslation(mSavedTranslations[id]);
+    setRotation(mSavedRotations[id]);
+  }
+}
+
+void WbTransform::save(const QString &id) {
+  WbGroup::save(id);
+  if (nodeModelName() != "TrackWheel") {
```

Comment:
Why don't you use a dynamic case instead of checking the model name?

Same comment for `WbTransform::reset`.

Reply:
I couldn't use the nodetype in the init because it wasn't set yet, and ended up doing the same in these functions but yes, don't need to check the model name in the reset/save.